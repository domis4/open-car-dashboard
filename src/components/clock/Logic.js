import { compose, lifecycle, withState } from 'recompose'
import moment from 'moment'

const withLifecycle = lifecycle({
  componentDidMount() {
    const { setTime } = this.props
    const updateClock = () =>
      setTimeout(() => {
        const now = moment().format('HH:mm')
        setTime(now)
        updateClock()
      }, 1000)
    updateClock()
  },
})

export default compose(
  withState('time', 'setTime', moment().format('HH:mm')),
  withLifecycle
)
