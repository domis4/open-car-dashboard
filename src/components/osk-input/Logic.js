import { withStateHandlers } from 'recompose'

handleFocus = () => {
  if (this.props.onFocus) {
    this.props.onFocus(this.input)
    this.setState({ inputNode: this.input })
    // the `this.refs.input` value should be passed to the Keyboard component as inputNode prop
  }
}

export default withStateHandlers(() => ({}), {
  handleFocus: () => () => ({ inputNode: this.input }),
})
