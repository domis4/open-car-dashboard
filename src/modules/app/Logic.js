import { compose, withState } from 'recompose'

export default compose(withState('isBootIconVisible', 'setBootIconVisible', true))
