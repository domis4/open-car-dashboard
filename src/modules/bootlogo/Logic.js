import { compose, lifecycle, withState } from 'recompose'

const withLifecycle = lifecycle({
  componentDidMount() {
    const { setFadeIn, callback, setShowElement, pathname } = this.props
    const timeout = pathname === '/' ? 2000 : 0

    setTimeout(() => {
      setFadeIn(false)
      setTimeout(() => setShowElement(false), 600)
      callback && setTimeout(() => callback(), 600)
    }, timeout)
  },
})

export default compose(
  withState('fadeIn', 'setFadeIn', true),
  withState('showElement', 'setShowElement', true),
  withLifecycle
)
