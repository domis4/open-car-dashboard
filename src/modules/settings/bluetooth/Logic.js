import { compose, withStateHandlers } from 'recompose'

export default compose(
  withStateHandlers({ isBluetoothOn: false }, () => ({
    activateBluetooth: () => () => {
      return { isBluetoothOn: true }
    },
  }))
)
