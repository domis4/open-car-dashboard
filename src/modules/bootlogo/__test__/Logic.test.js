import React from 'react'
import { compose } from 'recompose'
import { shallow } from 'enzyme'
import Logic from '../Logic'

describe('bootlogo module Logic.js', () => {
  it('expects component to match snapshot', () => {
    const Dummy = () => <div />
    const Component = compose(Logic)(Dummy)
    const wrapper = shallow(<Component />).dive()

    expect(wrapper).toMatchSnapshot()
  })

  it('expects component to receive a `setFadeIn` function as prop', () => {
    const Dummy = () => <div />
    const Component = compose(Logic)(Dummy)
    const wrapper = shallow(<Component />).dive()

    expect(wrapper.props().setFadeIn).toBeDefined()
  })
  it('expects component to receive a `setShowElement` function as prop', () => {
    const Dummy = () => <div />
    const Component = compose(Logic)(Dummy)
    const wrapper = shallow(<Component />).dive()

    expect(wrapper.props().setShowElement).toBeDefined()
  })
  it('expects component to receive a `fadeIn` prop', () => {
    const Dummy = () => <div />
    const Component = compose(Logic)(Dummy)
    const wrapper = shallow(<Component />).dive()

    expect(wrapper.props().fadeIn).toEqual(true)
  })
  it('expects component to receive a `showElement` prop', () => {
    const Dummy = () => <div />
    const Component = compose(Logic)(Dummy)
    const wrapper = shallow(<Component />).dive()

    expect(wrapper.props().showElement).toEqual(true)
  })
})
