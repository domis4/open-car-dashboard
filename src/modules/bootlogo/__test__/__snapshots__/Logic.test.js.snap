// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bootlogo module Logic.js expects component to match snapshot 1`] = `
<lifecycle(Dummy)
  fadeIn={true}
  setFadeIn={[Function]}
  setShowElement={[Function]}
  showElement={true}
/>
`;
